src/os/bluestore/BlueStore.cc

```diff
@@ -3650,6 +3656,10 @@ void BlueStore::ExtentMap::reshard(
   dout(20) << __func__ << "  extent_avg " << extent_avg << ", target " << target
 	   << ", slop " << slop << dendl;
 
+  uint32_t next_boundary = segment_size;
```

Comment:
May be move this and the next var definitions under if (onode_data_has_boundaries)?

Reply:
It could be done, but it will split declaration and initialization.
It wouldn't look nice.

Reply:
OK. Not a big deal at first glance.
I could name one benefit of having  the assignment below local though - one would be able to bypass that hypothetical division-by-zero if any by using zero segment size  - looks like we can disable usage of the one persistent at onode by setting per-store segment_size to 0.

This isn't used for segment_size == 0 scenario:
uint32_t encoded_segment_estimate = bytes * segment_size / (data_reshard_end - needs_reshard_begin);

Reply:
Yes, I agree now to split and fix division-by-zero.
Note - setting segment size to 0 will not work once onode is created. The segment_size is locked to metadata.

Reply:
I mean the ability to disable/enable onode segmentation using store's segment size parameter only. So looks like we've got this ability now. Hence I'm happy :)